plugin-server/tests/worker/ingestion/person-state.test.ts

```diff
@@ -2057,3 +2097,488 @@ describe('PersonState.update()', () => {
+
+    beforeAll(async () => {
+        ;[hub, closeHub] = await createHub({})
+    })
+
+    beforeEach(async () => {
+        const organizationId = await createOrganization(hub.db.postgres)
+        teamId = await createTeam(hub.db.postgres, organizationId)
+        // jest.useFakeTimers({ advanceTimers: 50 })
+    })
+
+    afterEach(() => {
+        jest.clearAllTimers()
+    })
+
+    afterAll(async () => {
+        await closeHub()
+    })
+
+    async function updatePersonStateFromEvent(event: Partial<PluginEvent>, ts = '', mergeAttempts = 3) {
+        const fullEvent = {
+            team_id: teamId,
+            properties: {},
+            ...event,
+        }
+        const t = ts ? DateTime.fromISO(ts).toUTC() : timestamp
+        const personContainer = new LazyPersonContainer(teamId, event.distinct_id!, hub)
+        const state = new PersonState(
+            fullEvent as any,
+            teamId,
+            event.distinct_id!,
+            t,
+            hub.db,
+            hub.statsd,
+            hub.personManager,
+            personContainer,
+            true,
+            undefined,
+            mergeAttempts
+        )
+        await state.update()
+    }
+
+    async function fetchPostgresPersons() {
+        const query = `SELECT * FROM posthog_person WHERE team_id = ${teamId} ORDER BY id`
+        return (await hub.db.postgres.query(query)).rows
+    }
+
+    async function fetchDistinctIds() {
+        const result = await hub.db.postgres.query(
+            `SELECT distinct_id, person_id FROM posthog_persondistinctid WHERE team_id=${teamId} ORDER BY id`
+        )
+        return result.rows.map(({ distinct_id, person_id }) => [distinct_id, person_id]).sort() as [string, string][]
+    }
+
+    async function fetchPersonIdOverrides() {
+        const result = await hub.db.postgres.query(
+            `
+            WITH overrides AS (
+                SELECT id, old_person_id, override_person_id
+                FROM posthog_personoverride
+                WHERE team_id = ${teamId}
+                ORDER BY id
+            )
+            SELECT
+                helper.uuid AS old_person_id,
+                overrides_helper.uuid AS override_person_id
+            FROM
+                overrides AS first
+            JOIN
+                posthog_personoverridehelper AS helper ON first.old_person_id = helper.id
+            JOIN (
+                SELECT
+                    second.id AS id,
+                    uuid
+                FROM
+                    overrides AS second
+                JOIN posthog_personoverridehelper AS helper ON second.override_person_id = helper.id
+            ) AS overrides_helper ON overrides_helper.id = first.id
+            `
+        )
+        return result.rows
+            .map(({ old_person_id, override_person_id }) => [old_person_id, override_person_id])
+            .sort() as [string, string][]
+    }
+
+    it('postgres overrides added on merge', async () => {
```

Comment:
Looks like there are no tests for CH side - can we add some

Reply:
What do you mean by "CH side"? Ingestion tests query ClickHouse.

Reply:
yes like the other tests earlier in this file - they verify that CH state is right too.

Reply:
Looks like it works in CI.